### ISSUE TITLE
Why GridSearchCV does not support XGBRanker?

### ISSUE THREAD

**Issue by ztphs980** [dmlc/xgboost#4844]
I have a piece of code to tune the parameters in XGBRanker using GridSearchCV. However, it keeps printing errors in "gsearch1.fit()" : Check failed: gptr.size() != 0 && gptr.back() == info.labels_.Size(): group structure not consistent with #rows.

The code is as follows:

    param_test1 = {
        'max_depth': range(3, 10, 2),
        'min_child_weight': range(1, 6, 2)
    }
    estimator = xgb.XGBRanker(learning_rate=0.1, n_estimators=3, max_depth=5,
                              min_child_weight=2, gamma=0, subsample=0.8, colsample_bytree=0.8,
                              objective='rank:pairwise', nthread=1, scale_pos_weight=1,
                              seed=27)
    estimator.fit(train_data, train_label, group=train_data_tem['cnt'])
    gsearch1 = GridSearchCV(estimator, param_grid=param_test1, scoring='average_precision', n_jobs=10, iid=False, cv=5)
    gsearch1.fit(X=train_data, y=train_label, **{'group':train_data_tem['cnt']})
    print(gsearch1.cv_results_, gsearch1.best_params_, gsearch1.best_score_)

Since "estimator.fit(train_data, train_label, group=train_data_tem['cnt'])" can run without errors, I guess that the format for "group" is correct. Can anyone tell me what's wrong in GridSearchCV? Thanks so much!

**Comment by hcho3** [comment]
I don't think your use case is supported, since GridSearchCV is not aware of `group` information at all. This is because none of the estimators in the scikit-learn package supports learning-to-rank task. (XGBRanker is from XGBoost package, not scikit-learn)
